src/test/java/org/jenkinsci/plugins/workflow/cps/ParamsVariableTest.java

```diff
@@ -65,7 +66,8 @@
         WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
         p.setDefinition(new CpsFlowDefinition("echo(/TEXT=${params.TEXT}/)",true));
         p.addProperty(new ParametersDefinitionProperty(new StringParameterDefinition("TEXT", "")));
-        r.assertLogContains("TEXT=null", r.assertBuildStatusSuccess(p.scheduleBuild2(0, new ParametersAction(new StringParameterValue("TEXT", /* not possible via UI, but to simulate other ParameterValue impls */null)))));
+        WorkflowRun b = r.assertBuildStatus(Result.FAILURE, p.scheduleBuild2(0, new ParametersAction(new StringParameterValue("TEXT", /* not possible via UI, but to simulate other ParameterValue impls */null))));
+        r.assertLogContains("Null value not allowed as an environment variable: TEXT", b);
```

Comment:
So in the previous behavior for this test, the code actually throws an `IllegalArgumentException` for the `echo` step at  this line: https://github.com/jenkinsci/workflow-cps-plugin/blob/87fe204710a8883ec854a995c2f973bd10ec7bab/src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java#L242
However, you can see that is in a `try/catch` so that exception gets swallowed here: https://github.com/jenkinsci/workflow-cps-plugin/blob/87fe204710a8883ec854a995c2f973bd10ec7bab/src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java#L248
The comment for the exception is talking about builds breaking because you cannot store a crazy  argument, which seems like that probably should not be the case for an `IllegalArgumentException`

Reply:
Thanks for explaining! I agree that it probably makes more sense to propagate this kind of exception. I was a little concerned that this change might break existing Pipelines, but I think that any step that actually used the environment variables would have failed anyway if they called `StepContext.get(EnvVars.class)` (right?), so it seems unlikely that there would be any Pipelines that rely on the previous behavior.